src/main/kotlin/com/demonwav/mcdev/platform/canary/CanaryModule.kt

```diff
@@ -31,23 +31,34 @@ import com.intellij.psi.search.GlobalSearchScope
 class CanaryModule<out T : AbstractModuleType<*>>(facet: MinecraftFacet, override val moduleType: T) : AbstractModule(facet) {
 
     override val type: PlatformType = moduleType.platformType
-    private var canaryInf: VirtualFile? = null
+    var canaryInf: VirtualFile? = null
+        get() {
+            if (field == null) {
```

Comment:
Another thought: Do the null checks for `canaryInf` and `neptuneInf` in `setup()` (and initialize them if needed) and then call `setup()` every time. That way we can reduce duplication...

Reply:
huh, thought I'd already done that :S